### DISCUSSION TITLE
Dataloader question

### DISCUSSION THREAD

**Discussion by ZIKEYUAN** [pyg-team/pytorch_geometric#3730] [Q&A] [answered]
Hi,
I have a question in Dataloader function. As we know, when we use Dataloader function, we will create some batch, and those graphs in each batch will seem like a big graph. For example , if we have two graphs G1 and G2, we feed them to a Dataloader function which batch_size = 2, if A1.edge_index = [[0,1],[1,0]] and A2.edge_index = [[0,1],[1,0]]. After this step, we will creat a Batch   B2 which include those two graphs and B2.edge_index = [[0,1,2,3],[1,0,3,2]], well its ture. But for other feature in the graph, such as node_type, A1.node_type = [0,1], A2.node_type = [0,1] this feature just for a single graph and not a global feture. So my problem is when I use Dataloader function, B2.node_type will is [0,1,0,1] but it is not correct , I need the feature seem like to edge_index and the B2.node_type is [0,1,2,3], so how to solve this question?

**Comment by rusty1s** [comment, marked as answer]
You can achieve this by customizing the `Data.__inc__` method, see [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/batching.html):
```python
class MyData(Data):
    def __inc__(self, key, value, *args, **kwargs):
        if key == 'node_type':
            return int(node_type.max()) + 1
        return super().__inc__(key, value, *args, **kwargs)
```

**Reply by ZIKEYUAN** [reply]
Thanks for your help!
